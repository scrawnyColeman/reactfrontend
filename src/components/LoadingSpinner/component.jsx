import React from 'react';
import spinner from '../../images/ajax-loader.gif';
import styled from 'styled-components';

const Container = styled.div`
    position: fixed;
    width: 100%;
    height: 100%;
    top: 0;
    left: 0;
`;
const StyledImg = styled.img`
    top: 30%;
    left: 48%;
    z-index: 1000;
    position: absolute;
`;

const LoadingSpinner = () => (
    <Container>
        <StyledImg src={spinner} alt="loading" />
    </Container>
);

export default LoadingSpinner;
