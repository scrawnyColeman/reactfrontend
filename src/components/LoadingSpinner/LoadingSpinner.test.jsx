import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import 'jest-styled-components';
import React from 'react';
import LoadingSpinner from './component';

Enzyme.configure({ adapter: new Adapter() });

describe('LoadingSpinner component', () => {
    test('component matches snapshot', () => {
        const wrapper = shallow(<LoadingSpinner />);
        expect(wrapper).toMatchSnapshot();
    });
});
