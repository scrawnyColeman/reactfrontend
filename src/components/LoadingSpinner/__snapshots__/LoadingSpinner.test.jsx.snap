// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingSpinner component component matches snapshot 1`] = `
<styled.div>
  <styled.img
    alt="loading"
    src="ajax-loader.gif"
    width="100px"
  />
</styled.div>
`;
